TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementCheckers.cs

```diff
@@ -16,10 +18,109 @@ public static class SavepointStatementChecker
                 if (savePointName.StartsWith("SYS", StringComparison.OrdinalIgnoreCase))
                 {
                     DiagnosticUtils.AddError(savepointStatement,
-                        "Invalid savepoint-name, it must not begin with 'SYS'.",
+                        "Invalid savepoint-name, it must not begin with 'SYS'.", context);
+                }
+            }
+        }
+    }
+
+    internal static class AlterSequenceStatementChecker
+    {
+        public static void OnCodeElement(AlterSequenceStatement alterSequenceStatement,
+            List<string> duplicatedClauses, bool emptyClauseSet,
+            CodeElementsParser.AlterSequenceStatementContext context)
+        {
+            foreach (var clauseType in duplicatedClauses)
+            {
+                DiagnosticUtils.AddError(alterSequenceStatement,
+                    clauseType + " cannot be specified more than once",
+                    context);
+            }
+
+            bool isInteger = false;
+            if (emptyClauseSet)
+            {
+                DiagnosticUtils.AddError(alterSequenceStatement,
+                    "At least one option among RESTART, INCREMENT, MINVALUE, MAXVALUE, CYCLE, CACHE or ORDER must be specified",
+                    context);
+            }
+
+            if (alterSequenceStatement.MinValue != null)
+            {
+                CheckIsInteger(alterSequenceStatement.MinValue.Literal);
+                if (!isInteger)
+                {
+                    DiagnosticUtils.AddError(alterSequenceStatement,
+                        " In minValue only digits '0' are allowed after decimal point",
```

Comment:
Factorize the diagnostic creation directly into `CheckIsInteger`, this will avoid problems like inconsistency in letter case or wrong messages.
For example:
- 'In minValue only digits '0' are allowed after decimal point' vs 'In RESTART value only digits '0' are allowed after decimal point', different letter case, always use uppercase for keywords appearing in diagnostic messages
- 'In RESTART value only digits '0' are allowed after decimal point' vs 'The INCREMENT value only digits '0' are allowed after decimal point', what happened ? The second message with 'The' is grammatically wrong.